public/plugin/bbb/lib/bbb.lib.php

```diff
@@ -747,55 +761,51 @@ public function getMeetingParticipantInfo($meetingId, $userId)
      *
      * @return false|int The last inserted ID. Otherwise return false
      */
-    public function saveParticipant($meetingId, $participantId)
+    public function saveParticipant(int $meetingId, int $participantId): false|int
     {
-        $meetingData = Database::select(
-            '*',
-            'plugin_bbb_room',
-            [
-                'where' => [
-                    'meeting_id = ? AND participant_id = ? AND close = ?' => [
-                        $meetingId,
-                        $participantId,
-                        BBBPlugin::ROOM_OPEN,
-                    ],
-                ],
-            ]
-        );
+        $em = Database::getManager();
 
-        foreach ($meetingData as $roomItem) {
-            $inAt = $roomItem['in_at'];
-            $outAt = $roomItem['out_at'];
-            $roomId = $roomItem['id'];
-            if (!empty($roomId)) {
-                if ($inAt != $outAt) {
-                    Database::update(
-                        'plugin_bbb_room',
-                        ['close' => BBBPlugin::ROOM_CLOSE],
-                        ['id = ? ' => $roomId]
-                    );
-                } else {
-                    Database::update(
-                        'plugin_bbb_room',
-                        ['out_at' => api_get_utc_datetime(), 'close' => BBBPlugin::ROOM_CLOSE],
-                        ['id = ? ' => $roomId]
-                    );
-                }
+        /** @var ConferenceActivityRepository $repo */
+        $repo = $em->getRepository(ConferenceActivity::class);
+
+        $meeting = $em->getRepository(ConferenceMeeting::class)->find($meetingId);
+        $user = $em->getRepository(User::class)->find($participantId);
```

Comment:
This returns an error and I do not know why.
`PHP Fatal error:  Uncaught Doctrine\\Persistence\\Mapping\\MappingException: Class 'Chamilo\\UserBundle\\Entity\\User' does not exist in /var/www/ofajmigdev.beeznest.com/www/vendor/doctrine/persistence/src/Persistence/Mapping/MappingException.php:80\nStack trace:\n#0`

To make it work I changed it with  :
`$user = api_get_user_entity($participantId);`

But I'm not sure it's a good idea to use a legacy function, it would be better to make it work with the repo.